### PR TITLE
Feat: card detection implementation

### LINKED ISSUE #56: Méthode : robot.detecter_carte() - (chapitre 5)
**

> Détecte si une carte est montré à la caméra.
> retourne le nom d'utilisateur de la carte détecté.

**

### Sur le code : https://github.com/42Angouleme/pybot
1. à implémenter (en français sur Robot.py)
2. à commenter dans le code (en français)
### Sur la doc : https://github.com/42Angouleme/42Angouleme.github.io
1. partie tutoriel dans son chapitre à rédiger
2. ajouter méthode dans la référence de la bibliothèque (ref.md)

### PR DESCRIPTION
**Pull request by ThibaudM13**
This PR is about implementing the method: `Robot.detecter_carte()`
- Some code have been deleted, because redundant or unused.
- Unachieved feature: Displaying name on camera output (Text displays vertically)
- Added method: `Robot.message_avertissement()`

This Pull Request should close #56

### CODE REVIEW

File: pybot/Robot.py [hunk 1]
```diff
@@ -213,13 +218,27 @@ def appliquer_filtre(self, chemin_fichier, nom_filtre):
...
+                - seuil_minimal (défaut: 0.75) : score minimum pour qu' une carte détectée soit considérée comme valide.
+                - seuil_arret_recherche (défaut: 0.85) : score pour qu' une carte détectée soit interprétée comme la bonne.
         """
-        return self.ecran.detect_card()
-    
+        if self.webapp is None:
+            self.message_avertissement(
+                "La fonction Robot.detecter_carte() a été appelée sans Robot.demarrer_webapp()")
+            return ""
+        users = self.ecran.detect_card()
```
Comment (gpoblon) [inline pybot/Robot.py:236]: detect_card should be in its own module, it should not be a _ecran_ module dependency

File: pybot/module_ecran/Ecran.py [hunk 2]
```diff
@@ -115,6 +119,6 @@ def display_image(self, file_path, x, y):
 
     def set_filter(self, file_path, filter_name):
         self.filters.apply(file_path, filter_name)
-    
+
     def detect_card(self):
```
Comment (gpoblon) [inline pybot/module_ecran/Ecran.py:123]: put this in the camera module

File: pybot/module_camera/UserCardsTracker.py [hunk 3]
```diff
@@ -3,24 +3,26 @@
...
+from ..module_webapp.models.user import UserResponse
+
+import pygame as pg
 
 
 class UserCardsTracker:
     def __init__(self, app: Flask):
         self.users, user_image_paths = self._get_users_info(app)
-        self.image_comparator = ImageComparator(user_image_paths)
+        self.image_comparator = ImageComparator(user_image_paths, sim_min_thresholds=(0.75, 0.85))
```
Comment (gpoblon) [inline pybot/module_camera/UserCardsTracker.py:25]: move this from the constructor to the draw method directly.

The draw method should take 2 optional parameters (thresholds).

File: pybot/Robot.py (hunk 1 above)
Comment (gpoblon) [reply pybot/Robot.py:236]: detect_card() should become connect() and return a full db user

Comment (gpoblon) [reply pybot/Robot.py:236]: add the supprimer_utilisateur() method that disconnect the **connected** user and removes it from the db

**Comment by ThibaudM13** [comment]
Should close all card/ session features: closes #57, closes #58, closes #59

**Review by gpoblon — APPROVED** [review]

**Pull request merged**